pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/AuthenticationDataBasic.java

```diff
@@ -42,9 +46,7 @@ public boolean hasDataForHttp() {
 
     @Override
     public Set<Map.Entry<String, String>> getHttpHeaders() {
-        Map<String, String> headers = new HashMap<>();
-        headers.put(HTTP_HEADER_NAME, httpAuthToken);
-        return headers.entrySet();
+        return Collections.unmodifiableMap(this.headers).entrySet();
```

Comment:
Why are you wrapping it all the times instead of creating this wrapper only once in the constructor? Here and in other places

Reply:
Thanks, fixed